bin/stt_test/stt_providers/ISTTBase.py

```diff
@@ -0,0 +1,69 @@
+import json
+import logging
+from typing import List, Literal, Protocol, TypedDict
+import speech_recognition as sr  # type: ignore
+import time
+from stt_test.utils.SentenceChecker import SentenceChecker
+from stt_test.utils.test_data import get_audio_with_transcription
+import inspect
+
+
+class STTResult(TypedDict):
+    stt_trans: str
+    trans: str
+    similarity: float
+    execution_time: float
+
+
+class ISSTBase(Protocol):
```

Comment:
What does "I" stand for?

Reply:
I wanted to mark it as some kind of interface

Reply:
I would like to stick to the convention with "I" (https://stackoverflow.com/a/10724816). I have no problems with other name changes. Would this suit you?

Reply:
Yes but let's not use the *Base suffix because it's a repetition.